# U.S. Medical Insurance Costs
Given dataset insurance costs and patients attributes that calculate to an insurance cost for patients. 

1. Our goal is to identify trends and identify patients attributes that incr/decr medical insurance costs. 

2. After reviewing the dataframe,  and sorting costs, we can see larger costs for smokers. Therefore will start by comparing the total cost in the database for smokers, and for non-smokers. 

3. We used mean and var functions to caculate the percentage of cost for each attibutes listed to see the proporations they contributed to total costs.  

4. Last we will use correlations of the numerical attributes of the patients lifestyle, to understand the direct and indirect linear relationship with charges. We used dummy variables in order to correlate the non-numerical answers for sex, smokers and region. 

In [20]:
#open csv data file
import pandas as pd
df_csv = pd.read_csv('insurance.csv')
df_csv.head()



age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

In [21]:
import pandas as pd

# Load the CSV file into a pandas dataframe
df = pd.read_csv('insurance.csv')

# Select the rows where the selected row is True
selected_rows = df.loc[df['smoker'] == 'yes']

# Calculate the sum, mean, and standard deviation of the 'numerical_column' for the selected rows
sum_of_charges = selected_rows['charges'].sum()
mean_of_charges = selected_rows['charges'].mean()
std_of_charges = selected_rows['charges'].std()

# Print the results of SMOKERS (yes)
print('Sum:', sum_of_charges)
print('Mean:', mean_of_charges)
print('Standard deviation:', std_of_charges)

Sum: 8781763.521839999
Mean: 32050.23183153284
Standard deviation: 11541.547175589121


In [22]:
import pandas as pd

# Load the CSV file into a pandas dataframe
df = pd.read_csv('insurance.csv')

# Select the rows where the 'boolean_column' is True
selected_rows = df.loc[df['smoker'] == 'no']

# Calculate the sum, mean, and standard deviation of the 'numerical_column' for the selected rows
sum_of_charges = selected_rows['charges'].sum()
mean_of_charges = selected_rows['charges'].mean()
std_of_charges = selected_rows['charges'].std()

# Print the results
print('Sum:', sum_of_charges)
print('Mean:', mean_of_charges)
print('Standard deviation:', std_of_charges)

Sum: 8974061.468919002
Mean: 8434.268297856204
Standard deviation: 5993.781819194933


In [23]:
import pandas as pd

# Load the CSV file into a pandas dataframe
df = pd.read_csv('insurance.csv')

# Select the rows where the 'boolean_column' is True
selected_rowsf = df.loc[df['sex'] == 'female']
selected_rowsm = df.loc[df['sex'] == 'male']

# Calculate the sum, mean, and standard deviation of the 'numerical_column' for the selected rows
sum_of_chargesf = selected_rowsf['charges'].sum()
mean_of_chargesf = selected_rowsf['charges'].mean()
std_of_chargesf = selected_rowsf['charges'].std()

# Print the results for non-Smokers
print('Sum:', sum_of_chargesf)
print('Mean:', mean_of_chargesf)
print('Standard deviation:', std_of_chargesf)

Sum: 8321061.194619
Mean: 12569.578843835347
Standard deviation: 11128.703800915431


In [32]:
import pandas as pd

# Load the CSV file into a pandas dataframe
df = pd.read_csv('insurance.csv')

# Calculate the mean and variance of the "charges" column for females and males, smokers and non-smokers,
# age levels, bmi levels and regions
age_young = df.loc[df['age'].between(0,25), 'charges'].mean()
age_medyoung = df.loc[df['age'].between(26,40), 'charges'].mean()
age_med = df.loc[df['age'].between(41,55), 'charges'].mean()
age_medold = df.loc[df['age'].between(56,63), 'charges'].mean()
higher_older = df.loc[df['age'] > 34 , 'charges'].mean()
low_bmi_mean = df.loc[df['bmi'] < 28 , 'charges'].mean()
higher_bmi_mean = df.loc[df['bmi'] > 27 , 'charges'].mean()
female_mean = df.loc[df['sex'] == 'female', 'charges'].mean()
male_mean = df.loc[df['sex'] == 'male', 'charges'].mean()
smokeryes_mean = df.loc[df['smoker'] == 'yes', 'charges'].mean()
smokerno_mean = df.loc[df['smoker'] == 'no', 'charges'].mean()
northwest_mean = df.loc[df['region'] == 'northwest', 'charges'].mean()
southwest_mean = df.loc[df['region'] == 'southwest', 'charges'].mean()
northeast_mean = df.loc[df['region'] == 'northeast', 'charges'].mean()
southeast_mean = df.loc[df['region'] == 'southeast', 'charges'].mean()

# Calculate the difference in mean and variance between males and females
sexmean_diff = male_mean - female_mean
sexmeanratio = male_mean/(male_mean+female_mean)
high_lowbmi_diff = higher_bmi_mean - low_bmi_mean
bmimeanratio = higher_bmi_mean/(higher_bmi_mean + low_bmi_mean)

smokermean_diff = smokeryes_mean - smokerno_mean
smokermeanratio = smokeryes_mean/(smokeryes_mean + smokerno_mean)

northwesteastmean_diff =  northwest_mean - northeast_mean
southwesteastmean_diff = southwest_mean - southeast_mean

# Print the results
print( "sexdiff = ", sexmean_diff, "smokerdiff = ", smokermean_diff, "northwesteastmean_diff = ", northwesteastmean_diff, "southwesteast_diff = ", southwesteastmean_diff)
print("sexmeanratio_male = ", sexmeanratio)
print("smokermeanratio_yes = ", smokermeanratio)
print("bmi_diff = ", high_lowbmi_diff)
print("bmi_meanratio = ", bmimeanratio)
print("age under 25 mean = ", age_young, "age between 26-40 mean = ",age_medyoung, "age between 41-55 mean = ", age_med, "age between 56-70 mean = ",age_medold)

sexdiff =  1387.1723338865468 smokerdiff =  23615.963533676637 northwesteastmean_diff =  -988.8091424165741 southwesteast_diff =  -2388.4740603175796
sexmeanratio_male =  0.5261470835347225
smokermeanratio_yes =  0.7916667299608452
bmi_diff =  3691.9691415861435
bmi_meanratio =  0.5735515311910594
age under 25 mean =  9087.015806924837 age between 26-40 mean =  11096.679994430378 age between 41-55 mean =  15515.617320261283 age between 56-70 mean =  18288.00508139175


sexdiff =  1387.17 Male average costs are greater than female avg cost. 
smokerdiff =  23615.96 People that smoke are much higher than average amount of non-smokers. 
northwesteastmean_diff =  -988.81 = Northwestern regions are lower in cost than northeast region. 
southwesteast_diff =  -2388.47 Southwest regions are lower than northeast region. 
sexmeanratio_male =  53% higher avg cost than women. 
smokermeanratio_yes =  80% higher avg cost than non-smokers.
bmi_diff =  3691.97 higher cost if their bmi > 27
bmi_meanratio =  57% higher cost with higher than >27 bmi. 
age under 25 mean =  9087.016 
age between 26-40 mean =  11096.69 
age between 41-55 mean =  15515.62
age between 56-70 mean =  18288.01

We want to see how closely correlated/relationships with the charges/cost of insurance. 

In [25]:
import pandas as pd

df = pd.read_csv('insurance.csv')

# Convert the "sex", "smoker", and "region" columns to dummy variables
df = pd.get_dummies(df, columns=['sex', 'smoker', 'region'])

correlations = df.corr()['charges']

# Print the correlations
print(correlations)

age                 0.299008
bmi                 0.198341
children            0.067998
charges             1.000000
sex_female         -0.057292
sex_male            0.057292
smoker_no          -0.787251
smoker_yes          0.787251
region_northeast    0.006349
region_northwest   -0.039905
region_southeast    0.073982
region_southwest   -0.043210
Name: charges, dtype: float64


Based on the above analysis of the file 'insurance.csv', we can conclude the following:
1. Whether a person is a smoker or not, weighed heavily and most directly to the higher or lower charges for insurance.
2. age at .299 was the next highest correlation, however not enough to be strong neither was bmi at .2 correlation. 
3. Although someone under 25 would be paying on average half the amount of someone over 56. 
4. Someone with a >27 bmi would be paying 57% higher on average than someone with bmi under 28. 
5. More notably someone that is smoker vs a non-smoker is paying almost 4x (400%) higher average costs for coverage than a non-smoker.